app/services/usps_in_person_proofing/enrollment_helper.rb

```diff
@@ -1,61 +1,94 @@
-        selected_location_details: selected_location_details,
-      )
-
-      enrollment_code = create_usps_enrollment(enrollment, pii)
-      return unless enrollment_code
-
-      # update the enrollment to status pending
-      enrollment.enrollment_code = enrollment_code
-      enrollment.status = :pending
-      enrollment.save!
-
-      send_ready_to_verify_email(user, pii, enrollment)
-    end
+    class << self
+      def schedule_in_person_enrollment(user, profile, pii, selected_location_details = nil)
+        enrollment = user.establishing_in_person_enrollment ||
+                     InPersonEnrollment.create!(user: user)
+
+        enrollment.profile = profile
+        enrollment.current_address_matches_id = pii['same_address_as_id']
+        if enrollment.selected_location_details.blank?
+          enrollment.selected_location_details = selected_location_details
+        end
+        enrollment.save!
+
+        enrollment_code = create_usps_enrollment(enrollment, pii)
+        return unless enrollment_code
+
+        # update the enrollment to status pending
+        enrollment.enrollment_code = enrollment_code
+        enrollment.status = :pending
+        enrollment.enrollment_established_at = Time.zone.now
+        enrollment.save!
 
-    def send_ready_to_verify_email(user, pii, enrollment)
-      user.confirmed_email_addresses.each do |email_address|
-        UserMailer.in_person_ready_to_verify(
-          user,
-          email_address,
-          first_name: pii['first_name'],
-          enrollment: enrollment,
-        ).deliver_now_or_later
+        send_ready_to_verify_email(user, pii, enrollment)
       end
-    end
 
-    def usps_proofer
-      if IdentityConfig.store.usps_mock_fallback
-        UspsInPersonProofing::Mock::Proofer.new
-      else
-        UspsInPersonProofing::Proofer.new
+      def send_ready_to_verify_email(user, pii, enrollment)
+        user.confirmed_email_addresses.each do |email_address|
+          UserMailer.in_person_ready_to_verify(
+            user,
+            email_address,
+            first_name: pii['first_name'],
+            enrollment: enrollment,
+          ).deliver_now_or_later
+        end
       end
-    end
 
-    def create_usps_enrollment(enrollment, pii)
-      applicant = UspsInPersonProofing::Applicant.new(
-        {
-          unique_id: enrollment.usps_unique_id,
-          first_name: pii['first_name'],
-          last_name: pii['last_name'],
-          address: pii['address1'],
-          # do we need address2?
-          city: pii['city'],
-          state: pii['state'],
-          zip_code: pii['zipcode'],
-          email: 'no-reply@login.gov',
-        },
-      )
-      proofer = usps_proofer
-
-      response = proofer.request_enroll(applicant)
-      response['enrollmentCode']
+      def establishing_in_person_enrollment_for_user(user)
+        cancel_stale_establishing_enrollments_for_user(user)
```

Comment:
I think we'll need to be able to cancel these earlier in the flow, before the desktop session initiates the hybrid flow. Since otherwise when the desktop session starts polling for a result, it may think the user has entered IPP due to a previous enrollment, since the hybrid session hasn't gotten far enough for the cancellations to happen.

Maybe that just means we expose `cancel_stale_establishing_enrollments_for_user` as a public method and we can call somewhere like [`WelcomeStep#call`](https://github.com/18F/identity-idp/blob/main/app/services/idv/steps/welcome_step.rb).